python/Sign_Extender.py

```diff
@@ -4,12 +4,12 @@
 def Sign_Extender(clk, instruction, output):
     #Instruction should be a 16-bit intbv
     #output is accurate 32-bit intbv.
-    output = intbv(max = 32)
+    output.Next = intbv(min = 32, max = 32)
     
     @alway(clk.posedge)
     def execute():
         #Insert 16 to 32-bit conversion loginc
-
-        return
+        output[31:16].Next = instruciton[15] #make the first new bits match the first bit
```

Comment:
Python doesn't allow arrays to be assigned a single value in batch. You might need to do a for loop ala
```
for out in output[16:31]:
  out.next = ..."
```

Also it looks like instruction is mispelled.